components/tree/Tree.tsx

```diff
@@ -100,7 +100,7 @@ export interface TreeProps {
   /** 展开/收起节点时触发 */
   onExpand?: (expandedKeys: string[], info: AntTreeNodeExpandedEvent) => void | PromiseLike<any>;
   /** 点击复选框触发 */
-  onCheck?: (checkedKeys: string[], e: AntTreeNodeCheckedEvent) => void;
+  onCheck?: (checkedKeys: string[] | { checked: string[]; halfChecked: string[] }, e: AntTreeNodeCheckedEvent) => void;
```

Comment:
it feels like we should make a change in rc-tree, instead of doing:
```
(checkedKeys: string[] | { checked: string[]; halfChecked: string[] }
```
We should keep it consistent to 
```
{ checked: string[]; halfChecked: string[] }
```
Just pass empty array to halfChecked if necessary.

@zombieJ